### PR TITLE
Update bootstrap to rustc 1.45.0-beta.2 (1dc0f6d8e 2020-06-15)

### LINKED ISSUE #73286: "./x.py test --stage 0 --doc" does not run doc tests
I would expect the following command to run the libcore doc tests:
```
./x.py test --stage 0 src/libcore --doc
```
This used to work, but now all it does it build libcore and its test binaries. No tests are actually being run (which I verified by breaking one of them deliberately).

@Mark-Simulacrum any idea what is happening?

### PR DESCRIPTION
**Pull request by ehuss**
Pulls in changes from #73326.

Closes #73286

### CODE REVIEW

**Comment by mati865** [comment]
You might want to wait on decision whether https://github.com/rust-lang/rust/pull/73184 is also accepted before updating bootstrap.

**Comment by ehuss** [comment]
Broken by #73447.
@bors r=Mark-Simulacrum p=1
Raising priority to avoid further breakage.

File: src/libcore/result.rs [hunk 1]
```diff
@@ -1189,12 +1190,15 @@ impl<T: DerefMut, E> Result<T, E> {
     /// # Examples
     ///
     /// ```
+    /// #![feature(inner_deref)]
+    /// let mut s = "HELLO".to_string();
     /// let mut x: Result<String, u32> = Ok("hello".to_string());
```
Comment (lzutao) [inline src/libcore/result.rs:1195]: Was wondering why CI doesn't catch this?

Comment (RalfJung) [reply src/libcore/result.rs:1195]: CI failed to run doctests at all. This PR fixes that. See https://github.com/rust-lang/rust/issues/73286.

Comment (lzutao) [reply src/libcore/result.rs:1195]: Thanks. I should had read the link issue.

**Pull request merged**